gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/OracleExtractor.java

```diff
@@ -0,0 +1,297 @@
+    commands.add(JdbcExtractor.getCommand(fetchSize, JdbcCommand.JdbcCommandType.FETCHSIZE));
+    return commands;
+  }
+
+  @Override
+  public Map<String, String> getDataTypeMap() {
+    return DATA_TYPE_MAP;
+  }
+
+  @Override
+  public Iterator<JsonElement> getRecordSetFromSourceApi(String schema, String entity, WorkUnit workUnit,
+      List<Predicate> predicateList) throws IOException {
+    return null;
+  }
+
+  @Override
+  public String getConnectionUrl() {
+    return String.format(URL_FORMAT, this.workUnit.getProp(ConfigurationKeys.SOURCE_CONN_HOST_NAME).trim(),
+                                     this.workUnit.getPropAsInt(ConfigurationKeys.SOURCE_CONN_PORT),
+                                     this.workUnit.getProp(ConfigurationKeys.SOURCE_QUERYBASED_SCHEMA).trim());
+  }
+
+  @Override
+  public long exractSampleRecordCountFromQuery(String query) {
+    if (StringUtils.isEmpty(query)) {
+      return -1;
+    }
+
+    String inputQuery = query.toLowerCase();
+    int idx = inputQuery.indexOf(SAMPLE_QUERY_KEYWORD);
+    if (idx < 0) {
+      return -1;
+    }
+
+    StringBuilder sb = new StringBuilder();
+    idx += SAMPLE_QUERY_KEYWORD.length();
+    for(; idx < inputQuery.length(); idx++) {
+      char c = inputQuery.charAt(idx);
+      if(c >= '0' && c <= '9') {
+        sb.append(c);
+      }
+    }
+
+    String sampleNumber = sb.toString();
+    if(StringUtils.isEmpty(sampleNumber)) {
+      LOG.warn("was not able to extract sample number on the query: " + query);
+      return -1;
+    }
+    return Long.parseLong(sampleNumber);
+
+  }
+
+  /**
+   * Purpose of this method is to remove sample clause like limit (from MySQL) or top (from SQL server)
+   * so that it can parse the query properly. As sample clause in Oracle is in where clause, it is not needed
+   * to remove it as it won't break any parsing process. (Removing condition from where clause is not simple and
+   * it's better to avoid when possible)
+   *
+   * {@inheritDoc}
+   * @see gobblin.source.extractor.extract.jdbc.JdbcSpecificLayer#removeSampleClauseFromQuery(java.lang.String)
+   */
+  @Override
+  public String removeSampleClauseFromQuery(String query) {
+    return query;
+  }
+
+  /**
+   * Purpose of this method is to add back sample clause. As it has not been removed from the first place
+   * ( @see removeSampleClauseFromQuery ), it does not need to add back. (Adding back condition to where clause is not simple and
+   * it's better to avoid when possible)
+   * {@inheritDoc}
+   * @see gobblin.source.extractor.extract.jdbc.JdbcSpecificLayer#constructSampleClause()
+   */
+  @Override
+  public String constructSampleClause() {
+    return "";
+  }
+
+  @Override
+  public String getWatermarkSourceFormat(WatermarkType watermarkType) {
+    String columnFormat = null;
+    switch (watermarkType) {
+      case TIMESTAMP:
```

Comment:
there should be a `break` clause for each `case`, otherwise FindBugs will throw a warning, which will cause the build to fail